test/concatAll.js

```diff
@@ -0,0 +1,30 @@
+import { assert } from 'chai';
+
+import { concatAll } from '../src';
+import eq from './shared/eq';
+
+describe('concatAll', function() {
+  it('concatenates arrays', function() {
+    eq(concatAll([[1], [2], [3]]), [1, 2, 3]);
+  });
+
+  it('concatenates strings', function() {
+    eq(concatAll(['1', '2', '3']), '123');
+  });
+
+  it('returns null if empty foldable was passed', function() {
+    eq(concatAll([]), null);
+  });
+
+  it('throws if non-foldable is passed', function() {
+    assert.throws(() => concatAll(null), TypeError);
```

Comment:
I don't see any tests with Semigroups.

Use `RA.Identity` or monet to create couple of tests; see here https://github.com/char0n/ramda-adjunct/blob/a3109fc1020c6f4a8686d032cb7d539a230af258/test/notBoth.js#L52

Reply:
I think I'll test it on [`NonEmptyList`](https://monet.github.io/monet.js/#non-empty-lists)

Reply:
Try I'm looking forward to see it

Reply:
Done